### DISCUSSION TITLE
ArnoldSetParameter Syntax

### DISCUSSION THREAD

**Discussion by Ark-Dem** [Autodesk/maya-usd#3582] [Q&A] [answered]
What is the proper syntax to make ArnoldSetParameter work?

I've tried this, but no luck:

```
def Mesh "pSphere1"
{
    ...
}

def ArnoldSetParameter "ArnoldSetParameter1"
{
	string arnold:selection = "/pSphere1"
	string[] arnold:assignment = ["subdiv_iterations=3", "subdiv_type='catclark'"]
	bool arnold:enable = 1
	bool[] arnold:enable_assignment = [1, 1]
}
```

**Comment by cpichard** [comment]
Hi @Ark-Dem ,
if you know the arnold syntax you can always write you set_parameter in a ass file then do
`kick mytest.ass -resave mytest.usda`
but from here your syntax looks correct

**Comment by sebastienblor** [comment]
indeed if you're used to working with operators in Arnold, you can setup a maya scene that works, and then export it to usd to see how it's supposed to be set.
In case you don't know, the operators graph needs to be assigned to the options, so in a scene that just contains what's in your snippet, the operator will be ignored.

**Comment by sebastienblor** [comment]
actually, how are you rendering this usd file ? are you rendering it directly with `kick scene.usd`, are you loading it back in a proxy shape, or are you loading it in another rendering application, like Solaris / usd_view, etc.. ?

**Comment by Ark-Dem** [comment]
@cpichard , nice trick, thanks.
@sebastienblor , I'm loading usda back into Maya as stage.

Besides options, operators can be connected to standins as well... So, I think I'm starting to understand the problem - since usd mimics maya/arnold scene, I can't connect an operator to Mesh prim... Only to ArnoldProcedural prim (or options).

It's just the current workflow is somewhat cumbersome - Arnold-USD creates extra attributes for subdiv parameters on Mesh prim, that usually ends up in binary .usd (so, I can't change subdiv_iterations by editing the file). I can't edit this attribute in stage because the attribute is not editable for some reasons... if I "edit/duplicate as maya object", all arnold attributes revert to default on maya object and I have to set them manually again... So, at the moment if I want to change subdiv_iterations for an asset, the best option ends up being re-exporting this asset from the original maya scene...

Or maybe there's a better approach?

**Reply by sebastienblor** [reply, marked as answer]
Ok, I understand that the source issue here is that this attribute isn't editable, which shouldn't be the case. It seems to happen because the attribute is declared as a byte attribute, but in practice we'd want it to be exposed as an integer. I'll check what can be done to fix that.

**Reply by Ark-Dem** [reply]
Yes, that should solve this problem.

Meanwhile, I've figured out that it already works by adding an override upstream manually. But, I guess, it won't be "uchar" type anymore soon...
```
def Xform "sphere" (
    prepend payload = @sphere.usda@
)
{
    over "pSphere1"
    {
        uchar primvars:arnold:subdiv_iterations = 3
    }
}
```
